src/database/statistics/spreadUtil.ts

```diff
@@ -71,110 +94,76 @@ export function getSpreadDates(
+      );
+    }
+    for (const month in spreadsPerDevs[dev].monthSpread) {
+      const repoArr = spreadsPerDevs[dev].monthSpread[month];
+      spreads.monthSpread = isRepoIDinDevSpreadArr(
+        repoArr,
+        spreads.monthSpread,
+        repoId,
+        dev,
+        month,
+      );
     }
   }
-  // add all amount of categorys to dates object
-  // this are all days, weeks, ... which are taken into account for that repo
-  dates.days = Object.keys(dates.daySpread).length;
-  dates.weeks = Object.keys(dates.weekSpread).length;
-  dates.sprints = Object.keys(dates.sprintSpread).length;
-  dates.months = Object.keys(dates.monthSpread).length;
-
-  return dates;
+  // sum of all time category items which are
+  // taken into account for that repo
+  spreads.days = Object.keys(spreads.daySpread).length;
+  spreads.weeks = Object.keys(spreads.weekSpread).length;
+  spreads.sprints = Object.keys(spreads.sprintSpread).length;
+  spreads.months = Object.keys(spreads.monthSpread).length;
+  return spreads;
 }
 
 /**
- * Helper function to eliminate duplicate weeks, sprints or months
- * because the calculated intervals per developer may differ
- * and would have been taken as separate intervals into account,
- * altough it was the same interval of commiting.
- * @param timeSpreadPairs An object with timestamps as keys, and an object with developer:sprad pairs as a value
- * @param days The days, the interval blongs to, i.e. 7 (weeks), 14 (sprint), 30(month)
- * @returns
+ * Check, if the @param repoId is in the @param repoArr of
+ * the current @param dev. If there is no entry for that
+ * @param date yet, add it to the @param spreadObj.
+ * Then, append the dev spread to that object.
+ * @returns spreadObj
  */
-export function rearangeTimeslots(
-  timeSpreadPairs: { [key: string]: { [key: string]: number } },
-  days: number,
-): { [key: string]: { [key: string]: number } } {
-  // copy the old timeSpreadPairs to modify on that
-  const newTimeSpreadPairs: { [key: string]: { [key: string]: number } } = {
-    ...timeSpreadPairs,
-  };
-  const timestamps: string[] = Object.keys(timeSpreadPairs).sort();
-  // go through every sorted timestamp and look to successor
-  for (let i = 1; i <= timestamps.length; ) {
-    const currentDate: string = timestamps[i - 1];
-    const nextDate: string = timestamps[i];
-    // check, if the next interval date should actualy be in current interval
-    if (addDays(currentDate, days) > new Date(timestamps[i])) {
-      // append the values together, update the current interval object
-      const devSpreadObj1 = timeSpreadPairs[currentDate];
-      const devSpreadObj2 = timeSpreadPairs[nextDate];
-      const mergedObj = { ...devSpreadObj1, ...devSpreadObj2 };
-      newTimeSpreadPairs[currentDate] = mergedObj;
-      // delete the unnessacary date and skip the next date
-      delete newTimeSpreadPairs[nextDate];
-      i += 2;
-    } else {
-      i += 1;
+function isRepoIDinDevSpreadArr(
```

Comment:
This name is confusing. I expect a function `isSomething` to check if its parameter fulfills some condition. I wouldn't expect it to have side effects like initializing an object.

Reply:
Yeah, right. I changed it to `createSetAndAddRepoIds()` and I am referring with that to the calculation which you suggested down below.